Debug build WebGL tests are slow because they output a lot of result text

kkinnunen-apple opened the pull request:
#### 138bf2852e8ab88f3a4bcd6418dab1807fa11091
<pre>
Debug build WebGL tests are slow because they output a lot of result text
<a href="https://bugs.webkit.org/show_bug.cgi?id=254736">https://bugs.webkit.org/show_bug.cgi?id=254736</a>
rdar://107415745

Reviewed by Alexey Proskuryakov.

Load the WebGL conformance tests in quiet mode. This avoids adding the results
to the DOM inside the iframe. WebKit StringView implementation is slowing the tests
with large amount of result prints on Debug. This causes excessive timeouts.

The results are shown for tests that complete via the wrapping test harness.
The results are not shown if the test has a syntax error. These should be debuggable
simply without test result lines in the bot results. In mysterious cases, the quiet
parameter can be removed from individual test runner wrappers temporarily.

* LayoutTests/http/tests/webgl/1.0.x/conformance/more/functions/readPixelsBadArgs.html:
* LayoutTests/http/tests/webgl/1.0.x/conformance/more/functions/texImage2DHTML.html:
* LayoutTests/http/tests/webgl/1.0.x/conformance/more/functions/texSubImage2DHTML.html:
* LayoutTests/http/tests/webgl/1.0.x/conformance/textures/misc/origin-clean-conformance-offscreencanvas.html:
* LayoutTests/http/tests/webgl/1.0.x/conformance/textures/misc/origin-clean-conformance.html:
* LayoutTests/http/tests/webgl/2.0.y/conformance/more/functions/readPixelsBadArgs.html:
* LayoutTests/http/tests/webgl/2.0.y/conformance/more/functions/texImage2DHTML.html:
* LayoutTests/http/tests/webgl/2.0.y/conformance/more/functions/texSubImage2DHTML.html:
* LayoutTests/http/tests/webgl/2.0.y/conformance/textures/misc/origin-clean-conformance-offscreencanvas.html:
* LayoutTests/http/tests/webgl/2.0.y/conformance/textures/misc/origin-clean-conformance.html:
* LayoutTests/http/tests/webgl/2.0.y/conformance2/textures/misc/origin-clean-conformance-offscreencanvas.html:
* LayoutTests/http/tests/webgl/resources/webkit-webgl-test-harness.js:
(window.webglTestHarness.reportResults):
(window.webglTestHarness.notifyFinished):
(list): Deleted.
* LayoutTests/webgl/1.0.x/conformance/attribs/gl-bindAttribLocation-aliasing.html:
* LayoutTests/webgl/1.0.x/conformance/attribs/gl-bindAttribLocation-matrix.html:
* LayoutTests/webgl/1.0.x/conformance/attribs/gl-bindAttribLocation-nonexistent-attribute.html:

... a lot of tests omitted ...

* LayoutTests/webgl/2.0.y/deqp/functional/gles3/vertexarrays/single_attribute.output_type.unsigned_byte.html:
* LayoutTests/webgl/2.0.y/deqp/functional/gles3/vertexarrays/single_attribute.output_type.unsigned_int.html:
* LayoutTests/webgl/2.0.y/deqp/functional/gles3/vertexarrays/single_attribute.output_type.unsigned_int_2_10_10_10.html:
* LayoutTests/webgl/2.0.y/deqp/functional/gles3/vertexarrays/single_attribute.output_type.unsigned_short.html:
* LayoutTests/webgl/2.0.y/deqp/functional/gles3/vertexarrays/single_attribute.stride.html:
* LayoutTests/webgl/2.0.y/deqp/functional/gles3/vertexarrays/single_attribute.usage.dynamic_copy.html:
* LayoutTests/webgl/2.0.y/deqp/functional/gles3/vertexarrays/single_attribute.usage.dynamic_draw.html:
* LayoutTests/webgl/2.0.y/deqp/functional/gles3/vertexarrays/single_attribute.usage.dynamic_read.html:
* LayoutTests/webgl/2.0.y/deqp/functional/gles3/vertexarrays/single_attribute.usage.static_copy.html:
* LayoutTests/webgl/2.0.y/deqp/functional/gles3/vertexarrays/single_attribute.usage.static_draw.html:
* LayoutTests/webgl/2.0.y/deqp/functional/gles3/vertexarrays/single_attribute.usage.static_read.html:
* LayoutTests/webgl/2.0.y/deqp/functional/gles3/vertexarrays/single_attribute.usage.stream_copy.html:
* LayoutTests/webgl/2.0.y/deqp/functional/gles3/vertexarrays/single_attribute.usage.stream_draw.html:
* LayoutTests/webgl/2.0.y/deqp/functional/gles3/vertexarrays/single_attribute.usage.stream_read.html:
* Tools/Scripts/webkitpy/update_webgl_conformance_tests_lib/main.py:
* Tools/Scripts/webkitpy/update_webgl_conformance_tests_lib/webgl-expectation-template.txt:
* Tools/Scripts/webkitpy/update_webgl_conformance_tests_lib/webgl-test-driver-template.html:
* Tools/Scripts/webkitpy/update_webgl_conformance_tests_lib/webkit-webgl-test-harness-template.js:
(window.webglTestHarness.reportResults):
(window.webglTestHarness.notifyFinished):
(list): Deleted.

Canonical link: <a href="https://commits.webkit.org/262387@main">https://commits.webkit.org/262387@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/60ba3bba81f7ccbbf9a0ac781ca8b841e29e4cc2

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/1254 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/1288 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/1328 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/2049 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/1123 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/1341 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/1357 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/1282 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/1264 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/1183 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/1171 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/1912 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/1198 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/1159 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/1135 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/1149 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/1190 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/2247 "run-api-tests-without-change (failure)") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/1197 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/1096 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/1134 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/1169 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/349 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/1224 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

aproskuryakov reviewed (approved):
Are those debug assertions slowing it down, or just lack of optimization? If it's the former, have you considered removing those? They likely make other tests slower, even if not so dramatically.

kkinnunen-apple committed:
Debug build WebGL tests are slow because they output a lot of result text
https://bugs.webkit.org/show_bug.cgi?id=254736
rdar://107415745

Reviewed by Alexey Proskuryakov.

Load the WebGL conformance tests in quiet mode. This avoids adding the results
to the DOM inside the iframe. WebKit StringView implementation is slowing the tests
with large amount of result prints on Debug. This causes excessive timeouts.

The results are shown for tests that complete via the wrapping test harness.
The results are not shown if the test has a syntax error. These should be debuggable
simply without test result lines in the bot results. In mysterious cases, the quiet
parameter can be removed from individual test runner wrappers temporarily.

* LayoutTests/http/tests/webgl/1.0.x/conformance/more/functions/readPixelsBadArgs.html:
* LayoutTests/http/tests/webgl/1.0.x/conformance/more/functions/texImage2DHTML.html:
* LayoutTests/http/tests/webgl/1.0.x/conformance/more/functions/texSubImage2DHTML.html:
* LayoutTests/http/tests/webgl/1.0.x/conformance/textures/misc/origin-clean-conformance-offscreencanvas.html:
* LayoutTests/http/tests/webgl/1.0.x/conformance/textures/misc/origin-clean-conformance.html:
* LayoutTests/http/tests/webgl/2.0.y/conformance/more/functions/readPixelsBadArgs.html:
* LayoutTests/http/tests/webgl/2.0.y/conformance/more/functions/texImage2DHTML.html:
* LayoutTests/http/tests/webgl/2.0.y/conformance/more/functions/texSubImage2DHTML.html:
* LayoutTests/http/tests/webgl/2.0.y/conformance/textures/misc/origin-clean-conformance-offscreencanvas.html:
* LayoutTests/http/tests/webgl/2.0.y/conformance/textures/misc/origin-clean-conformance.html:
* LayoutTests/http/tests/webgl/2.0.y/conformance2/textures/misc/origin-clean-conformance-offscreencanvas.html:
* LayoutTests/http/tests/webgl/resources/webkit-webgl-test-harness.js:
(window.webglTestHarness.reportResults):
(window.webglTestHarness.notifyFinished):
(list): Deleted.
* LayoutTests/webgl/1.0.x/conformance/attribs/gl-bindAttribLocation-aliasing.html:
* LayoutTests/webgl/1.0.x/conformance/attribs/gl-bindAttribLocation-matrix.html:
* LayoutTests/webgl/1.0.x/conformance/attribs/gl-bindAttribLocation-nonexistent-attribute.html:

... a lot of tests omitted ...

* LayoutTests/webgl/2.0.y/deqp/functional/gles3/vertexarrays/single_attribute.output_type.unsigned_byte.html:
* LayoutTests/webgl/2.0.y/deqp/functional/gles3/vertexarrays/single_attribute.output_type.unsigned_int.html:
* LayoutTests/webgl/2.0.y/deqp/functional/gles3/vertexarrays/single_attribute.output_type.unsigned_int_2_10_10_10.html:
* LayoutTests/webgl/2.0.y/deqp/functional/gles3/vertexarrays/single_attribute.output_type.unsigned_short.html:
* LayoutTests/webgl/2.0.y/deqp/functional/gles3/vertexarrays/single_attribute.stride.html:
* LayoutTests/webgl/2.0.y/deqp/functional/gles3/vertexarrays/single_attribute.usage.dynamic_copy.html:
* LayoutTests/webgl/2.0.y/deqp/functional/gles3/vertexarrays/single_attribute.usage.dynamic_draw.html:
* LayoutTests/webgl/2.0.y/deqp/functional/gles3/vertexarrays/single_attribute.usage.dynamic_read.html:
* LayoutTests/webgl/2.0.y/deqp/functional/gles3/vertexarrays/single_attribute.usage.static_copy.html:
* LayoutTests/webgl/2.0.y/deqp/functional/gles3/vertexarrays/single_attribute.usage.static_draw.html:
* LayoutTests/webgl/2.0.y/deqp/functional/gles3/vertexarrays/single_attribute.usage.static_read.html:
* LayoutTests/webgl/2.0.y/deqp/functional/gles3/vertexarrays/single_attribute.usage.stream_copy.html:
* LayoutTests/webgl/2.0.y/deqp/functional/gles3/vertexarrays/single_attribute.usage.stream_draw.html:
* LayoutTests/webgl/2.0.y/deqp/functional/gles3/vertexarrays/single_attribute.usage.stream_read.html:
* Tools/Scripts/webkitpy/update_webgl_conformance_tests_lib/main.py:
* Tools/Scripts/webkitpy/update_webgl_conformance_tests_lib/webgl-expectation-template.txt:
* Tools/Scripts/webkitpy/update_webgl_conformance_tests_lib/webgl-test-driver-template.html:
* Tools/Scripts/webkitpy/update_webgl_conformance_tests_lib/webkit-webgl-test-harness-template.js:
(window.webglTestHarness.reportResults):
(window.webglTestHarness.notifyFinished):
(list): Deleted.

Canonical link: https://commits.webkit.org/262387@main
